lib/native_enum/activerecord_enum_post7.rb

```diff
@@ -0,0 +1,69 @@
+module ActiveRecord
+  module ConnectionAdapters
+    if defined?(AbstractMysqlAdapter)
+      class AbstractMysqlAdapter
+        class << self
```

Comment:
Change here was to change the `initialize_type_map_with_enum`, `register_enum_type` and `register_set_type` methods to be class methods instead of instance methods.